core/common/privdata/membershipinfo.go

```diff
@@ -7,36 +7,97 @@ SPDX-License-Identifier: Apache-2.0
 package privdata
 
 import (
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
 	"github.com/hyperledger/fabric/common/flogging"
 	"github.com/hyperledger/fabric/msp"
 	"github.com/hyperledger/fabric/protos/common"
+	mspp "github.com/hyperledger/fabric/protos/msp"
+	"github.com/pkg/errors"
 )
 
 var logger = flogging.MustGetLogger("common.privdata")
 
 // MembershipProvider can be used to check whether a peer is eligible to a collection or not
 type MembershipProvider struct {
+	mspID                       string
 	selfSignedData              common.SignedData
 	IdentityDeserializerFactory func(chainID string) msp.IdentityDeserializer
 }
 
 // NewMembershipInfoProvider returns MembershipProvider
-func NewMembershipInfoProvider(selfSignedData common.SignedData, identityDeserializerFunc func(chainID string) msp.IdentityDeserializer) *MembershipProvider {
-	return &MembershipProvider{selfSignedData: selfSignedData, IdentityDeserializerFactory: identityDeserializerFunc}
+func NewMembershipInfoProvider(mspID string, selfSignedData common.SignedData,
+	identityDeserializerFunc func(chainID string) msp.IdentityDeserializer) *MembershipProvider {
+	return &MembershipProvider{mspID: mspID, selfSignedData: selfSignedData, IdentityDeserializerFactory: identityDeserializerFunc}
 }
 
 // AmMemberOf checks whether the current peer is a member of the given collection config.
 // If getPolicy returns an error, it will drop the error and return false - same as a RejectAll policy.
-func (m *MembershipProvider) AmMemberOf(channelName string, collectionPolicyConfig *common.CollectionPolicyConfig) (bool, error) {
+// It is used when a chaincode is upgraded to see if the peer's org has become eligible after	a collection
```

Comment:
A nit. Not sure whether we need to comment about the usages of a function. We have tools/editors that help in finding all the callers/callee/references. I am saying this because we use this function twice during a chaincode upgrade.
1.  to check whether the peer is already a member (using the existing collection policy);
2.  to check whether the peer is a newly added member (using the new policy if the check during (1) returns false).

Using both (1) and (2), we check whether a peer has become a new member of a collection. I understand that this comment was added as a result of some questions raised during the review (when `AmMemberOf` and `isEligible` are used). Hence, I am fine if we plan to retain it.

Reply:
I added it in response to Dave's request for additional clarification above. I can move it down so it's not in the function signature comment and rather within the function if that helps to clarify the specific use case.

Reply:
I've confused between AmMemberOf and isEligible, so I'm fine to keep it.